libdirections-onboard/src/main/java/com/mapbox/navigation/route/onboard/MapboxOnboardRouter.kt

```diff
@@ -30,11 +32,13 @@ import kotlinx.coroutines.withContext
  * tiles' version, token. Config is provided via [MapboxOnboardRouterConfig].
  *
  * @param navigatorNative Native Navigator
+ * @param logger interface for logging any events
  */
 @MapboxModule(MapboxModuleType.NavigationOnboardRouter)
 class MapboxOnboardRouter(
     private val navigatorNative: MapboxNativeNavigator,
-    config: MapboxOnboardRouterConfig
+    config: MapboxOnboardRouterConfig,
+    private val logger: Logger?
```

Comment:
Do we want this to be nullable 🤔? Probably yes but the `Logger` created in `MapboxNavigation` is non-nullable https://github.com/mapbox/mapbox-navigation-android/blob/1b5f5fb7ea54f3ecf01335b461026de5f0bc3bd1/libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt#L141

Same question applies to other classes 👇

Reply:
Good question, @Guardiola31337 ! But as I understand, users of our SDK can use `MapboxOnboardRouter` directly without `MapboxNavigation`, right? So in this case, I think, we need to give them possibility to not pass `logger`, wdyt?

Reply:
it's not really true, users _cannot_ and _shouldn't_ use any `Route` directly and must interrupt with our components via `MapboxNavigation` only

Reply:
No one is expected to use the default module implementations standalone, but even if they do, they can also take the default logger module implementation and push it in. Or implement their own using the interface definition.

In short, I think we can make it non-nullable.

Reply:
👍 Thanks for clarification!